lib/builder_deferred_tagging.rb

```diff
@@ -1,6 +1,17 @@
 require "builder_deferred_tagging/version"
+require 'builder/xmlmarkup'
 
 module BuilderDeferredTagging
-  class Error < StandardError; end
-  # Your code goes here...
+  Builder::XmlMarkup.class_eval do
+    @@deferred_attributes = {}
```

Comment:
I'm not sure this is going to work as a class-level variable? I can see the upside in that I can use this in ActsAsRDFable without needing access to the specific instance being used in OAIsys buuuuuuut this map is only going to get initialized _once_, when Ruby loads this module for the first time. On the _second_ request to the API, this map will still be populated with whatever contents it built up from the first request, will it not?

Reply:
When serving the response xml.deferred_attributes (below) is called which clears the deferred_attributes hash

Reply:
Ah, yeah, that makes sense.

Fuzzier issue, though: by using a class variable, this gem isn't actually _thread safe_ because the class state would be shared across threads in the same process in a thread-per-request model (I dug this up this morning, it's actually a really good read: https://bearmetal.eu/theden/how-do-i-know-whether-my-rails-app-is-thread-safe-or-not/).

Now, right now, in Jupiter, we're using Passenger open-source edition in Pro AFAIK, so we're actually doing process-per-request and not thread-per-request (and I haven't checked, but we could have other threading issues in Jupiter as well), but I feel like we're backing ourselves into a corner if we switch to a threaded request model for scalability, if we knowingly create threading issues, and it might make this gem less useful for anyone else that wants to pick it up?

Reply:
I understand the risk of using class variables making this gem not thread safe. Is it worth approaching this problem another way to make it thread safe? I'm not sure how the problem could be approached or if it would be worth it. What do you think?